doc/lsst.source.injection/reference/1_make_injection_pipeline.rst

```diff
@@ -0,0 +1,167 @@
+Typically, a pipeline reference file will be a pipeline that is being used to reduce data as part of a data reduction campaign.
+
+Either the :doc:`make_injection_pipeline <../scripts/make_injection_pipeline>` command line script or the associated :py:func:`~lsst.source.injection.make_injection_pipeline` Python function may be used to generate a fully qualified injection pipeline.
+Examples on this page illustrate the use of both methods.
+
+.. _lsst.source.injection-reference-make-injection-pipeline-stubs:
+
+Injection Pipeline Stubs
+=========================
+
+A number of different source injection pipeline stubs have been constructed in the ``$SOURCE_INJECTION_DIR/pipelines`` directory.
+Each of these pipeline stubs contain a single task that is used to inject sources into a particular dataset type.
+
+Although these injection pipeline YAML stubs can be used directly, it is recommended that the :doc:`make_injection_pipeline <../scripts/make_injection_pipeline>` command line script or the associated :py:func:`~lsst.source.injection.make_injection_pipeline` Python function be used to generate a complete source injection pipeline definition YAML file for subsequent use.
+A complete pipeline definition file will contain the pipeline stub as a subtask, along with any additional tasks required to complete the source injection process.
+
+The table below lists the available pipeline YAML stubs inside the ``$SOURCE_INJECTION_DIR/pipelines`` directory and the dataset types they are designed to inject sources into:
+
+.. list-table::
+   :widths: auto
+   :header-rows: 1
+
+   * - Injection Dataset Type
+     - Injection Pipeline Stub
+     - Injection Task
+   * - postISRCCD
+     - inject_exposure.yaml_
+     - :lsst-task:`~lsst.source.injection.ExposureInjectTask`
+   * - calexp
+     - inject_visit.yaml_
+     - :lsst-task:`~lsst.source.injection.VisitInjectTask`
+   * - deepCoadd
+     - inject_coadd.yaml_
+     - :lsst-task:`~lsst.source.injection.CoaddInjectTask`
+
+.. _inject_exposure.yaml: https://github.com/lsst/source_injection/blob/main/pipelines/inject_exposure.yaml
+.. _inject_visit.yaml: https://github.com/lsst/source_injection/blob/main/pipelines/inject_visit.yaml
+.. _inject_coadd.yaml: https://github.com/lsst/source_injection/blob/main/pipelines/inject_coadd.yaml
+
+.. _lsst.source.injection-reference-make-injection-pipeline-command-line:
+
+Making an Injection Pipeline on the Command Line
+================================================
+
+The :doc:`make_injection_pipeline <../scripts/make_injection_pipeline>` command line script is used to generate a complete source injection pipeline definition YAML file.
+More information on the operation of this script may be obtained by running ``make_injection_pipeline --help``.
+
+An example calling this script from the command line:
+
+.. code-block:: shell
+
+    $SOURCE_INJECTION_DIR/bin/make_injection_pipeline \
```

Comment:
Yes, *but*, you'd need to also provide a butler repo for it to query against. This task at present requires no actual butler repo to exist, which makes it rather simple. We could remove the `-i` argument, but we'd need to replace it with a `-b` argument. I'm not sure which is more obvious?

Side note: you don't _have_ to specify `-i` in this command for it to succeed. If `-i` is not passed as an argument, this script will simply amend all the connection type names after `postISRCCD` to be prefixed with `injected_`. This may or may not be a desirable feature, but it is an option available at present.

Reply:
Following a discussion on Zoom about this today, I've opened this ticket: [DM-39748](https://jira.lsstcorp.org/browse/DM-39748). I think we can implement this change seamlessly, whilst still maintaining the current behavior. The ticket above will explore this in more detail.